=== lib/Plagger/Plugin/Store/DBIC.pm
==================================================================
--- lib/Plagger/Plugin/Store/DBIC.pm  (revision 2182)
+++ lib/Plagger/Plugin/Store/DBIC.pm  (local)
@@ -42,7 +42,7 @@
 
     $self->conf->{schema_class}->require
         or $c->error(
-        qq/Can't load schema class "@{[ $self->conf->{schema_class} ]}", $!/);
+        qq/Can't load schema class "@{[ $self->conf->{schema_class} ]}", $@/);
 
     $self->schema( $self->conf->{schema_class}
             ->connect( @{ $self->conf->{connect_info} } ) );
@@ -63,9 +63,19 @@
             type => $args->{feed}->type,
         }
     );
-    for (qw/url image description language author updated/) {
-        $feed->$_( $args->{feed}->$_ ) if !$feed->$_ or $feed->$_ ne $args->{feed}->$_;
+
+    my $data;
+    for my $attr (qw/url title description language author updated/) {
+        $data->{$attr} = $args->{feed}->$attr if defined $args->{feed}->$attr;
     }
+
+    # xxx
+    $data->{image} = $args->{feed}->image->{url} if $args->{feed}->image;
+    $data->{updated} ||= $args->{feed}->entries->[0]->date || Plagger::Date->now;
+
+    for my $attr (keys %$data) {
+        $feed->$attr( $data->{$attr} ) if !$feed->$attr or $feed->$attr ne $data->{$attr};
+    }
     $feed->insert_or_update if !$feed->in_storage or $feed->is_changed;
 
     # feed meta
@@ -93,10 +103,9 @@
             link => $feed_entry->link,
         });
 
-        for (qw/title author summary date body rate/) {    # todo: icon
-            $entry->$_( $feed_entry->$_ )
-                if !$entry->$_
-                or $entry->$_ ne $feed_entry->$_;
+        for my $attr (qw/title author summary date body rate/) {    # todo: icon
+            $entry->$attr( $feed_entry->$attr )
+                if !$entry->$attr or $entry->$attr ne $feed_entry->$attr;
         }
         $entry->insert_or_update if $entry->is_changed;
 
@@ -115,7 +124,7 @@
             if @{ $entry_diff->{deleted} };
 
         for ( @{ $entry_diff->{added} } ) {
-            $entry->add_to_tags( { name => $_ } );
+            $entry->add_to_tags( { name => $_ } ) if $entry->id; # xxx
         }
     }
 }
=== lib/Plagger/Schema/SQLite/Entry.pm
==================================================================
--- lib/Plagger/Schema/SQLite/Entry.pm  (revision 2182)
+++ lib/Plagger/Schema/SQLite/Entry.pm  (local)
@@ -8,7 +8,7 @@
 
 
 __PACKAGE__->add_columns(
-    qw/id feed title author date link summary body rate icon/
+    qw/id feed title author date link summary body rate icon read/
 );
 __PACKAGE__->set_primary_key('id');
 __PACKAGE__->utf8_columns(qw/title author link summary body rate icon/);
=== lib/Plagger/Schema/SQLite/Feed.pm
==================================================================
--- lib/Plagger/Schema/SQLite/Feed.pm  (revision 2182)
+++ lib/Plagger/Schema/SQLite/Feed.pm  (local)
@@ -8,13 +8,21 @@
 
 
 __PACKAGE__->add_columns(
-    qw/id link url image description language author updated type/
+    qw/id link url title image description language author updated type/
 );
 __PACKAGE__->set_primary_key('id');
-__PACKAGE__->utf8_columns(qw/link url image description language author type/);
+__PACKAGE__->utf8_columns(qw/link url title image description language author type/);
 
 __PACKAGE__->has_many( entries => 'Plagger::Schema::SQLite::Entry', 'feed' );
 __PACKAGE__->has_many( tags  => 'Plagger::Schema::SQLite::FeedTag',  'feed' );
 __PACKAGE__->has_many( metas => 'Plagger::Schema::SQLite::FeedMeta', 'feed' );
 
+__PACKAGE__->inflate_column(
+    updated => {
+        inflate =>
+            sub { DateTime->from_epoch( epoch => shift, time_zone => 'local' ) },
+        deflate => sub { shift->set_time_zone('local')->epoch },
+    }
+);
+
 1;
=== sql/plagger.sqlite.sql
==================================================================
--- sql/plagger.sqlite.sql  (revision 2182)
+++ sql/plagger.sqlite.sql  (local)
@@ -2,6 +2,7 @@
        id INTEGER NOT NULL PRIMARY KEY,
        link TEXT,
        url TEXT,
+       title TEXT,
        image TEXT,
        description TEXT,
        language TEXT,
@@ -20,7 +21,8 @@
        summary TEXT,
        body TEXT,
        rate TEXT,
-       icon TEXT
+       icon TEXT,
+       read INTEGER NOT NULL DEFAULT 0
 );
 
 CREATE TABLE feed_meta (
